### ISSUE TITLE
Unit-based duration format behaviour differs from documentation

### ISSUE THREAD

**Issue by InverseIntegral** [neo4j/neo4j#12861]
The documentation on duration values specifies that the unit-based format expects `n` to denote an arbitrarily large number:

> The n denotes a numeric value which can be arbitrarily large.

When I use large numbers, however, I get exceptions. Moreover, the exceptions seem to differ depending on the value that I supply.

**Neo4j Version:** 4.4.5
**Operating System:** 5.16.16 Arch
**API:** Java

### Expected behavior
No exception should be thrown according to the specification.

### Actual behavior
A `NumberFormatException` and a `QueryExcecutionException` are thrown.

### First query
```
CREATE ({prop: duration('P1754475719991321997M')})
```
```
org.neo4j.graphdb.QueryExecutionException: Invalid value for duration, will cause overflow. Value was months=1754475719991321997, days=0, seconds=0, nanos=0
	at org.neo4j.kernel.impl.query.QueryExecutionKernelException.asUserException(QueryExecutionKernelException.java:35)
	at org.neo4j.cypher.internal.javacompat.ResultSubscriber.converted(ResultSubscriber.java:426)
	at org.neo4j.cypher.internal.javacompat.ResultSubscriber.assertNoErrors(ResultSubscriber.java:407)
	at org.neo4j.cypher.internal.javacompat.ResultSubscriber.init(ResultSubscriber.java:85)
	at org.neo4j.cypher.internal.javacompat.ExecutionEngine.executeQuery(ExecutionEngine.java:116)
	at org.neo4j.kernel.impl.coreapi.TransactionImpl.execute(TransactionImpl.java:285)
	at org.neo4j.kernel.impl.coreapi.TransactionImpl.execute(TransactionImpl.java:274)
	at org.neo4j.kernel.impl.factory.GraphDatabaseFacade.executeTransactionally(GraphDatabaseFacade.java:164)
	at org.neo4j.kernel.impl.factory.GraphDatabaseFacade.executeTransactionally(GraphDatabaseFacade.java:154)
	at org.neo4j.kernel.impl.factory.GraphDatabaseFacade.executeTransactionally(GraphDatabaseFacade.java:142)
```

### Second query
```
CREATE ({prop: duration('P18446744073709551615M')})
```
```
java.lang.NumberFormatException: For input string: "18446744073709551615"
	at java.base/java.lang.NumberFormatException.forInputString(NumberFormatException.java:65)
	at java.base/java.lang.Long.parseLong(Long.java:692)
	at java.base/java.lang.Long.parseLong(Long.java:817)
	at org.neo4j.values.storable.DurationValue.optLong(DurationValue.java:531)
	at org.neo4j.values.storable.DurationValue.parseDuration(DurationValue.java:376)
	at org.neo4j.values.storable.DurationValue.parse(DurationValue.java:342)
	at org.neo4j.values.storable.TemporalValue.parse(TemporalValue.java:428)
	at org.neo4j.values.storable.DurationValue.parse(DurationValue.java:140)
	at org.neo4j.procedure.impl.temporal.DurationFunction.apply(DurationFunction.java:99)
	at org.neo4j.procedure.impl.ProcedureRegistry.callFunction(ProcedureRegistry.java:250)
	at org.neo4j.procedure.impl.GlobalProceduresRegistry.callFunction(GlobalProceduresRegistry.java:358)
	at org.neo4j.kernel.impl.newapi.AllStoreHolder.callBuiltInFunction(AllStoreHolder.java:1146)
	at org.neo4j.kernel.impl.newapi.AllStoreHolder.builtInFunctionCall(AllStoreHolder.java:1037)
	at org.neo4j.cypher.internal.runtime.interpreted.CallSupport$.callBuiltInFunction(CallSupport.scala:42)
	at org.neo4j.cypher.internal.runtime.interpreted.TransactionBoundReadQueryContext.callBuiltInFunction(TransactionBoundQueryContext.scala:1153)
	at org.neo4j.cypher.internal.planning.ExceptionTranslatingReadQueryContext.callBuiltInFunction(ExceptionTranslatingQueryContext.scala:241)
	at org.neo4j.cypher.internal.runtime.interpreted.DelegatingQueryContext.callBuiltInFunction(DelegatingQueryContext.scala:397)
	at org.neo4j.cypher.internal.runtime.interpreted.commands.expressions.BuiltInFunctionInvocation.call(FunctionInvocation.scala:51)
	at org.neo4j.cypher.internal.runtime.interpreted.commands.expressions.FunctionInvocation.apply(FunctionInvocation.scala:42)
	at org.neo4j.cypher.internal.runtime.interpreted.commands.expressions.LiteralMap.$anonfun$apply$1(LiteralMap.scala:36)
	at scala.collection.immutable.Map$Map1.foreach(Map.scala:192)
	at org.neo4j.cypher.internal.runtime.interpreted.commands.expressions.LiteralMap.apply(LiteralMap.scala:35)
	at org.neo4j.cypher.internal.runtime.interpreted.pipes.BaseCreatePipe.setProperties(CreatePipe.scala:53)
	at org.neo4j.cypher.internal.runtime.interpreted.pipes.EntityCreatePipe.$anonfun$createNode$2(CreatePipe.scala:94)
	at org.neo4j.cypher.internal.runtime.interpreted.pipes.EntityCreatePipe.$anonfun$createNode$2$adapted(CreatePipe.scala:94)
	at scala.Option.foreach(Option.scala:407)
	at org.neo4j.cypher.internal.runtime.interpreted.pipes.EntityCreatePipe.createNode(CreatePipe.scala:94)
	at org.neo4j.cypher.internal.runtime.interpreted.pipes.CreatePipe.$anonfun$internalCreateResults$2(CreatePipe.scala:138)
	at org.neo4j.cypher.internal.runtime.interpreted.pipes.CreatePipe.$anonfun$internalCreateResults$2$adapted(CreatePipe.scala:137)
	at scala.collection.IndexedSeqOptimized.foreach(IndexedSeqOptimized.scala:36)
	at scala.collection.IndexedSeqOptimized.foreach$(IndexedSeqOptimized.scala:33)
	at scala.collection.mutable.ArrayOps$ofRef.foreach(ArrayOps.scala:198)
	at org.neo4j.cypher.internal.runtime.interpreted.pipes.CreatePipe.$anonfun$internalCreateResults$1(CreatePipe.scala:137)
	at org.neo4j.cypher.internal.runtime.ClosingIterator$$anon$3.next(ClosingIterator.scala:154)
	at org.neo4j.cypher.internal.runtime.interpreted.pipes.EmptyResultPipe.internalCreateResults(EmptyResultPipe.scala:30)
	at org.neo4j.cypher.internal.runtime.interpreted.pipes.PipeWithSource.createResults(Pipe.scala:82)
	at org.neo4j.cypher.internal.runtime.interpreted.pipes.PipeWithSource.createResults(Pipe.scala:78)
	at org.neo4j.cypher.internal.runtime.interpreted.PipeExecutionResult.request(PipeExecutionResult.scala:69)
	at org.neo4j.cypher.internal.result.StandardInternalExecutionResult.request(StandardInternalExecutionResult.scala:90)
	at org.neo4j.cypher.internal.result.StandardInternalExecutionResult.initiate(StandardInternalExecutionResult.scala:53)
	at org.neo4j.cypher.internal.result.ClosingExecutionResult.$anonfun$initiate$1(ClosingExecutionResult.scala:58)
	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
	at org.neo4j.cypher.internal.result.ClosingExecutionResult.safely(ClosingExecutionResult.scala:99)
	at org.neo4j.cypher.internal.result.ClosingExecutionResult.initiate(ClosingExecutionResult.scala:58)
	at org.neo4j.cypher.internal.result.ClosingExecutionResult$.wrapAndInitiate(ClosingExecutionResult.scala:183)
	at org.neo4j.cypher.internal.CypherCurrentCompiler$CypherExecutableQuery.innerExecute(CypherCurrentCompiler.scala:442)
	at org.neo4j.cypher.internal.CypherCurrentCompiler$CypherExecutableQuery.execute(CypherCurrentCompiler.scala:379)
	at org.neo4j.cypher.internal.ExecutionEngine.doExecute(ExecutionEngine.scala:217)
	at org.neo4j.cypher.internal.ExecutionEngine.$anonfun$executeSubquery$1(ExecutionEngine.scala:178)
	at org.neo4j.cypher.internal.ExecutionEngine.closing(ExecutionEngine.scala:183)
	at org.neo4j.cypher.internal.ExecutionEngine.executeSubquery(ExecutionEngine.scala:175)
	at org.neo4j.cypher.internal.ExecutionEngine.execute(ExecutionEngine.scala:125)
	at org.neo4j.cypher.internal.javacompat.ExecutionEngine.executeQuery(ExecutionEngine.java:127)
	at org.neo4j.cypher.internal.javacompat.ExecutionEngine.executeQuery(ExecutionEngine.java:115)
	at org.neo4j.kernel.impl.coreapi.TransactionImpl.execute(TransactionImpl.java:285)
	at org.neo4j.kernel.impl.coreapi.TransactionImpl.execute(TransactionImpl.java:274)
	at org.neo4j.kernel.impl.factory.GraphDatabaseFacade.executeTransactionally(GraphDatabaseFacade.java:164)
	at org.neo4j.kernel.impl.factory.GraphDatabaseFacade.executeTransactionally(GraphDatabaseFacade.java:154)
	at org.neo4j.kernel.impl.factory.GraphDatabaseFacade.executeTransactionally(GraphDatabaseFacade.java:142)
```

**Comment by HannesSandberg** [comment]
Thank you for the repport! We will look into this and come back to you.

**Comment by JonasHolgersson** [comment]
Sorry for the late update: The docs were updated to show the actual max value of a duration, and the confusing exception was fixed in release 5.1
